From dd224dcb2c7fee323e45e00bfde6e72d1b5595ba Mon Sep 17 00:00:00 2001
From: mbah37 <mbah37@student.cscc.edu>
Date: Wed, 12 Nov 2025 16:17:01 -0500
Subject: [PATCH 08/17] 3rd tutorial. set up a dragon class and displayed ship
 on screen(will display dragon on screen when I get the images)

---
 alien_invasion.py |  6 ++++++
 dragon.py         | 24 ++++++++++++++++++++++++
 settings.py       |  8 ++++++--
 3 files changed, 36 insertions(+), 2 deletions(-)
 create mode 100644 dragon.py

diff --git a/alien_invasion.py b/alien_invasion.py
index 07b8f8d..d1135e9 100644
--- a/alien_invasion.py
+++ b/alien_invasion.py
@@ -1,6 +1,7 @@
 import sys
 import pygame
 from settings import Settings
+from dragon import Dragon
 
 class WhiteWalkerInvasion:
 
@@ -19,6 +20,9 @@ class WhiteWalkerInvasion:
 
         self.running = True
         self.clock = pygame.time.Clock()
+
+        # change variable to dragon. remove comment when done
+        self.ship = Dragon(self)
     
     def run_game(self):
         while self.running:
@@ -28,6 +32,8 @@ class WhiteWalkerInvasion:
                     pygame.quit()
                     sys.exit()
             self.screen.blit(self.bg, (0, 0))
+            # change variable to dragon. remove comment when done
+            self.ship.draw()
             pygame.display.flip()
             self.clock.tick(self.settings.FPS)
 
diff --git a/dragon.py b/dragon.py
new file mode 100644
index 0000000..941f9d4
--- /dev/null
+++ b/dragon.py
@@ -0,0 +1,24 @@
+import pygame
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from alien_invasion import WhiteWalkerInvasion
+
+class Dragon:
+    
+    def __init__(self, game: 'WhiteWalkerInvasion'):
+        self.game = game
+        self.settings = game.settings
+        self.screen = game.screen
+        self.screen_rect = self.screen.get_rect()
+
+        #change variable to dragon for height and width. remove comment when done
+        self.image = pygame.image.load(self.settings.ship_file)
+        self.image = pygame.transform.scale(self.image,
+            (self.settings.ship_width, self.settings.ship_height))
+        
+        self.rect = self.image.get_rect()
+        self.rect.midbottom = self.screen_rect.midbottom
+
+    def draw(self):
+        self.screen.blit(self.image, self.rect)
\ No newline at end of file
diff --git a/settings.py b/settings.py
index 561cf25..2f55f83 100644
--- a/settings.py
+++ b/settings.py
@@ -4,8 +4,12 @@ class Settings:
     def __init__(self):
         self.name : str = 'White Walker Invasion'
         self.screen_width : int = 1200
-        self.screen_height : int = 800
+        self.screen_height : int = 700
         self.FPS : int = 60
         self.bg_file : Path = Path.cwd() / 'Assets' / 'images' / 'Winterfell_background.png'
 
-
+        #change variable to dragon_file and change path and other variables.
+        # remove comments when done
+        self.ship_file : Path = Path.cwd() / 'Assets' / 'images' / 'ship2(no bg).png'
+        self.ship_width : int = 40
+        self.ship_height : int = 70
-- 
2.51.0.windows.1

